pgmpy/models/DynamicBayesianNetwork.py

```diff
@@ -440,6 +440,51 @@ def remove_cpds(self, *cpds):
                 cpd = self.get_cpds(cpd)
             self.cpds.remove(cpd)
 
+    def get_cardinality(self, node=None):
+        """
+        Returns the cardinality of the node.
+
+        Parameter
+        ---------
+        node: tuple (node_name, time_slice)
+            The node should be in the following form (node_name, time_slice).
+            Here, node_name is the node that is inserted while the time_slice is
+            an integer value, which denotes the index of the time_slice that the
+            node belongs to.
+
+        Returns
+        -------
+        int or dict : If node is specified returns the cardinality of the node.
+                      If node is not specified returns a dictionary with the given
+                      variable as keys and their respective cardinality as values.
+
+        Examples:
+        -------
+        >>> from pgmpy.models import DynamicBayesianNetwork as DBN
+        >>> from pgmpy.factors.discrete import TabularCPD
+        >>> dbn = DBN()
+        >>> dbn.add_edges_from([(('D',0),('G',0)),(('I',0),('G',0)),(('D',0),('D',1)),(('I',0),('I',1))])
+        >>> grade_cpd =  TabularCPD(('G',0), 3, [[0.3,0.05,0.9,0.5],
+        ...                                      [0.4,0.25,0.8,0.03],
+        ...                                      [0.3,0.7,0.02,0.2]], [('I', 0),('D', 0)],[2,2])
+        >>> dbn.add_cpds(grade_cpd)
+        >>> dbn.get_cardinality(('D',0))
+        2
+        >>> dbn.get_cardinality()
+        defaultdict(int, {('D', 0): 2, ('D', 1): 2, ('G', 0): 3, ('I', 0): 2, ('I', 1): 2})
+        """
+        if node:
+            if node not in super(DynamicBayesianNetwork, self).nodes():
```

Comment:
I think this will work only when the time slice is 0 or 1. Also add tests for testing when this method will throw this error.